### ISSUE TITLE
Assertion violation at src/tsolvers/UFTHandler.cc:38 (interpolants)

### ISSUE THREAD

**Issue by rainoftime** [usi-verification-and-security/opensmt#130]
Hi, for the following QF_UF formula,
[38.txt](https://github.com/usi-verification-and-security/opensmt/files/4803277/38.txt)

opensmt 9820755
~~~~
unsat
opensmt: /home/opensmt/src/tsolvers/UFTHandler.cc:38: virtual PTRef UFTHandler::getInterpolant(const ipartitions_t&, std::map<PTRef, opensmt::icolor_t>*): Assertion `iegraph' failed.
Aborted
~~~~

**Comment by rainoftime** [comment]
Another formula
[38_2.txt](https://github.com/usi-verification-and-security/opensmt/files/4803282/38_2.txt)

**Comment by blishko** [comment]
Hi Jack!

As written in the project README:

> When using OpenSMT as an executable, interpolation is off by default. It can be enabled by passing -i option to the executable, or by setting the SMT-LIB option produce-interpolants in the input file **before** the (set-logic) command.

So, you need to set the option before you set the logic.

We are aware that the error handling should be improved and we plan to work on that.

**Comment by rainoftime** [comment]
Thanks!  I met the problem for some other solver before (certain options should be set before `set-logic'`)

**Comment by blishko** [comment]
> When using OpenSMT as an executable, interpolation is off by default. It can be enabled by passing -i option to the executable, or by setting the SMT-LIB option produce-interpolants in the input file **before** the (set-logic) command.

OpenSMT will now report error and ignore the (set-option) command now when processing an SMT-LIB script where this condition would be violated.
This has been implemented as part of #139
